### PR TITLE
Allocation profiler and some profiler tweaks

### PR DESCRIPTION
**Pull request by pfitzseb**
image

- adds `@profview_allocs`/`view_alloc_profile`
- flame graph view of allocations is scaled with `x^1/5` because of the big differences in magnitude of allocation
- in-editor display is now always relative to root count
- percentage displayed in flame graph is now always relative to root count

### CODE REVIEW

File: scripts/packages/VSCodeServer/src/profiler.jl [hunk 1]
```diff
@@ -138,3 +137,105 @@ macro profview(ex, args...)
...
+
+"""
+    @profview_allocs f(args...) [sample_rate=0.0001] [C=false]
+
+Clear the Profile buffer, profile `f(args...)`, and view the result graphically.
+"""
+macro profview_allocs(ex, args...)
+    sample_rate_expr = :(sample_rate=0.0001)
+    for arg in args
+        if arg.head == :(=) && length(arg.args) > 0 && arg.args[1] == :sample_rate
```
Comment (aviatesk) [inline scripts/packages/VSCodeServer/src/profiler.jl:151]:
```suggestion
        if Meta.isexpr(arg, :(=)) && length(arg.args) > 0 && arg.args[1] === :sample_rate
```

File: scripts/packages/VSCodeServer/src/profiler.jl [hunk 2]
```diff
@@ -138,3 +137,105 @@ macro profview(ex, args...)
...
+                continue
+            end
+            file = string(sf.file)
+            this = get!(this[:children], hash(sf), Dict(
+                :func => sf.func,
+                :file => basename(file),
+                :path => fullpath(file),
+                :line => sf.line,
+                :count => 0,
+        :scaledCount => 0,
```
Comment (aviatesk) [inline scripts/packages/VSCodeServer/src/profiler.jl:193]:
```suggestion
                :count => 0,
                :scaledCount => 0,
```

File: scripts/packages/VSCodeServer/src/VSCodeServer.jl [hunk 3]
```diff
@@ -1,7 +1,7 @@
 module VSCodeServer
 
 export vscodedisplay, @vscodedisplay, @enter, @run
-export view_profile, @profview
+export view_profile, view_alloc_profile, @profview, @profview_allocs
```
Comment (aviatesk) [inline scripts/packages/VSCodeServer/src/VSCodeServer.jl:4]:
```suggestion
export view_profile, view_profile_allocs, @profview, @profview_allocs
```
?

**Comment by pfitzseb** [comment]
This now supports a `count` and an `allocs` mode, much like PProf:
image
image

**Comment by aviatesk** [comment]
Yay!

**Pull request merged**